src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,60 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+    struct Stats
+    {
+
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , peerDisconnects (collector->make_gauge("Overlay","Peer_Disconnects")) 
+            { 
+                mcollector=collector;
+            }
+
+        beast::insight::Hook hook;
+        beast::insight::Gauge peerDisconnects;
+        std::vector<beast::insight::Gauge> trafficGauges;
+        beast::insight::Collector::ptr mcollector;
+        
+    };
+    
+    Stats m_stats;
+
+private:
+    void collect_metrics()
+    {   
+        std::lock_guard lock (mutex_);
+
+        auto const stats = m_traffic.getCounts();
+
+        if(m_stats.trafficGauges.empty()){
+            for (auto const& i : stats)
+            {   
+                m_stats.trafficGauges.push_back(
+                    beast::insight::Gauge (m_stats.mcollector->make_gauge(i.name,"Bytes_In")));
+                m_stats.trafficGauges.push_back(
+                    beast::insight::Gauge (m_stats.mcollector->make_gauge(i.name,"Bytes_Out")));
+                m_stats.trafficGauges.push_back(
+                    beast::insight::Gauge (m_stats.mcollector->make_gauge(i.name,"Messages_In")));
+                m_stats.trafficGauges.push_back(
+                    beast::insight::Gauge (m_stats.mcollector->make_gauge(i.name,"Messages_Out")));
+            }
+        }
+
```

Comment:
It is not ideal for trafficGauges to be partially initialized in the constructor and completed here. Since this is a nested struct and m_traffic is a member of the parent class, we can't do `auto const stats = m_traffic.getCounts();` in the constructor. Another option might be to have a copy of the array of categories in the Stats struct. Please advise if there are any alternative workarounds. 

For reference, this is the original array of TrafficStats objects with category names:
https://github.com/ripple/rippled/blob/90d9ca901d06b9da1653526c234bc4b00e27070b/src/ripple/overlay/impl/TrafficCount.h#L182